### PR TITLE
clusterer: Update AgglomerativeClustering keyword to fix deprecation

### PR DESCRIPTION
**Pull request by EwoutH**
Changes the keyword for the metric used to compute the linkage in [sklearn.cluster.AgglomerativeClustering](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html) from `affinity`, which is deprecated in scikit-learn 1.2, to `metric`.

Behaviour should be identical.

See https://scikit-learn.org/stable/whats_new/v1.2.html#sklearn-cluster.

> API Change The affinity attribute is now deprecated for [cluster.AgglomerativeClustering](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html#sklearn.cluster.AgglomerativeClustering) and will be renamed to metric in v1.4. [#23470](https://github.com/scikit-learn/scikit-learn/pull/23470) by [Meekail Zain](https://github.com/micky774).

### CODE REVIEW

**Comment by EwoutH** [comment]
I got confirmation from scikit-learn that this doesn't change the behaviour, just a name change:

> Yes, you should be able to replace `affinity-->metric` with no change in behavior!

@quaquel I need your approval to merge (that's a new policy for this repo right?)

**Review by quaquel — APPROVED** [review]

**Comment by quaquel** [comment]
Thanks for checking.
